### ISSUE TITLE
upload-img-directly has some very strange and probably license violating code

### ISSUE THREAD

**Issue by GarboMuffin** [STForScratch/ScratchTools#365]
https://github.com/STForScratch/ScratchTools/blob/471e718369e9588ea84b288006ab82f1553dd8b3/features/upload-img-directly.js#L33-L38

I cannot imagine any reason this needs to be a `<script>` tag.

Furthermore, this code possibly violates at least two licenses simultaneously:

 - I presume that this code is not written by you and is probably under a license such as MIT which requires attribution (among other requirements). I do not see such attribution.
 - Per the GPL, I am entitled to receive a copy of this project's source code in the "preferred form of the work for making modifications to it". Without directions to find this, I believe this license is being violated.

**Comment by rgantzos** [comment]
@veloce-vasta Did you write this code yourself?

**Comment by veloce-vasta** [comment]
no, I found it on a post on scratch a long time ago, which had no license, and I credited my source

**Comment by GarboMuffin** [comment]
This issue specifically refers to line 35, which contains a JavaScript library compressed down to 9000 characters in a big string. I assure you that neither you nor the person who wrote upload-img-directly.js wrote this library because you wouldn't have minified it for no reason. Regardless of who wrote it, the minified code is not the "preferred form of the work for making modifications to it", so it does not satisfy the GPL's requirements.

The concern with attributing whichever Scratch user wrote the original version of upload-img-directly.js is a separate issue, not something I particularly care about as the code is not minified and you at least seem to credit some people in features.json.

*I am not a lawyer; this is not legal advice.*

**Comment by veloce-vasta** [comment]
We should probably just remove the whole feature to save us a lot of pain and anguish. i'm not about to track down the real owner either

**Comment by GarboMuffin** [comment]
or just replace the questionable code with something else. if it's just md5 then it won't be hard to find a replacement.

**Comment by rgantzos** [comment]
It's not a super complicated feature, I'll just remake it myself
